uwblueprint/social-diversity-for-children
# feat(admin): admin internal dashboard

### Notion ticket link



[Main Internal Dashboard Page](https://www.notion.so/uwblueprintexecs/Main-Internal-Dashboard-Page-1fd119199fbb4b9ebd5ba4e11a79825b)





https://user-images.githubusercontent.com/44826218/139561143-609cff07-7ef9-4a23-a58e-879d4ac2c88e.mp4


### Implementation description

- Add page component under `/admin/index.tsx`
- This admin component uses new SWR queries to get stats and upcoming classes (along with any live classes)
- Uses existing getZoomLink SWR query to do join class
- The admin page will redirect to `/no-access` if user is not teacher or admin
- The stats route will only allow users who are teacher or admin to access



### Steps to test

1. Try to access `/admin` while logged in as parent - you should be redirected to a 404 page
2. Try to access `/admin` while logging teacher or admin - all stats should load correctly, upcoming and live classes are accurate, you can join the current live class



### What should reviewers focus on?

- Dev: Logic of get upcoming class backend is a bit janky especially for weekends, I think it's enough of an edge case but I'm open to ideas! 
- Design: Slight change to live class card - add volunteer count info (I feel like this would be important to admins)
- Design: I feel like the page is a bit squished, the dashboard header has not been added but when it does, we'll see how it goes 😅 . It will be taken care of in https://www.notion.so/uwblueprintexecs/Add-Dashboard-header-to-Admin-page-wrapper-211e9fa33291479dab24390078479e50 

### Checklist
